### DISCUSSION TITLE
Treemap Click Event

### DISCUSSION THREAD

**Discussion by Akdora** [d3/d3#3466] [Q&A] [answered]
Hi There,
How can I toggle nodes for v6.x? https://observablehq.com/@d3/tidy-tree
There are bunch of examples for v3.x, but it does not work. 
Thank you

**Comment by mbostock** [comment, marked as answer]
See https://observablehq.com/@d3/collapsible-tree

**Reply by Akdora** [reply]
I see the click part. d variable was set to event. There are actually two input variables. Thank you!!

`    const nodeEnter = node.enter().append("g")
        .on("click", (event, d) => {
          d.children = d.children ? null : d._children;
          update(d);
        });`

**Reply by Fil** [reply]
If you need a reference about this major change in d3v6, see https://observablehq.com/@d3/d3v6-migration-guide#events

**Reply by Junfeizhao** [reply]
I have a questing about v6. 
I get tow parms from click function ,the click event and the node, but the node missing something like x,x0,

**Reply by Junfeizhao** [reply]
different ways to generate nodes between v3 and v6:
v3: _nodes = _tree.nodes(_root).reverse();
v6: _nodes = treeRoot.descendants().reverse();

the nodes of v3  has property of x,but v6 do not has
